crypto/bn/bn_dsa.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <openssl/opensslconf.h>
+#ifdef FIPS_MODE
+NON_EMPTY_TRANSLATION_UNIT
+#else
+# include <stdio.h>
+# include <string.h>
+# include <openssl/evp.h>
+# include <openssl/rand_drbg.h>
+# include "internal/cryptlib.h"
+# include "crypto/bn.h"
+# include "internal/thread_once.h"
+# include "../rand/rand_local.h"
+
+typedef struct rfc6979_seed_st
+{
+    const unsigned char *x;
+    const unsigned char *h;
+    size_t rlen;
+} RFC6979_SEED;
+
+static size_t rfc6979_get_entropy(RAND_DRBG *drbg, unsigned char **pout,
+                          int entropy, size_t min_len, size_t max_len,
+                          int prediction_resistance)
+{
+    RFC6979_SEED *rfc6979_seed = RAND_DRBG_get_callback_data(drbg);
+
+    *pout = (unsigned char *)rfc6979_seed->x;
+    return rfc6979_seed->rlen;
+}
+
+static size_t rfc6979_get_nonce(RAND_DRBG *drbg, unsigned char **pout,
+                        int entropy, size_t min_len, size_t max_len)
+{
+    RFC6979_SEED *rfc6979_seed = RAND_DRBG_get_callback_data(drbg);
+
+    *pout = (unsigned char *)rfc6979_seed->h;
+    return rfc6979_seed->rlen;
+}
+
+static int bits2int(BIGNUM *out, int qlen,
+    const unsigned char *message, size_t message_len)
+{
+    if (BN_bin2bn(message, (int)message_len, out) == NULL)
+        return 0;
+    if ((int)message_len * 8 > qlen)
+        return BN_rshift(out, out, (int)message_len * 8 - qlen);
+    return 1;
+}
+
+static int int2octets(unsigned char *out, const BIGNUM *num, int rlen)
+{
+    return BN_bn2binpad(num, out, rlen);
+}
+
+static int bits2octets(unsigned char *out, const BIGNUM *range,
+    const unsigned char *message, size_t message_len, BN_CTX *ctx)
+{
+    BIGNUM *num = BN_new();
+
+    if (!bits2int(num, BN_num_bits(range), message, message_len)
+       || !BN_mod(num, num, range, ctx)
```

Comment:
indent this and the next line an additional 4 spaces.  use 4-space indents overall (except for parameters i guess)

Reply:
The 4 space extra indentation is optional. Personally, I don't add it (because @levitte's Emacs indentation style doesn't support it  ;-) ), but choose as _you_ prefer.

Anyway, at least one space needs to be added to align the beginning of the continued lines (here: the first `|`) with the content of the if condition (the exclamation mark):

```C
    if (!bits2int(num, BN_num_bits(range), message, message_len)
        || !BN_mod(num, num, range, ctx)
        || !BN_bn2binpad(num, out, BN_num_bytes(range)))
    {
        BN_clear_free(num);
        return 0;
    }
```